hack/generator/pkg/config/configuration.go

```diff
@@ -176,16 +176,63 @@ func (config *Configuration) ShouldPrune(typeName astmodel.TypeName) (result Sho
 // If no transformation is performed, nil is returned
 func (config *Configuration) TransformType(name astmodel.TypeName) (astmodel.Type, string) {
 	for _, transformer := range config.TypeTransformers {
-		result := transformer.TransformTypeName(name)
-		if result != nil {
-			return result, transformer.Because
+		if transformer.propertyRegex == nil { // exclude property transformers
+			result := transformer.TransformTypeName(name)
+			if result != nil {
+				return result, transformer.Because
+			}
 		}
 	}
 
 	// No matches, return nil
 	return nil, ""
 }
 
+// PropertyTransformResult is the result of applying a property type transform
+type PropertyTransformResult struct {
+	NewType         *astmodel.ObjectType
+	Property        astmodel.PropertyName
+	NewPropertyType astmodel.Type
+	Because         string
+}
+
+// TransformTypeProperties applies any property transformers to the type
+func (config *Configuration) TransformTypeProperties(name astmodel.TypeName, objectType *astmodel.ObjectType) *PropertyTransformResult {
+
+	for _, transformer := range config.TypeTransformers {
+		if transformer.propertyRegex != nil { // exclude non-property transformers
```

Comment:
Are these really two different transformers hidden in one?

Reply:
Maybe in initialize I split them into two sets?

Reply:
Done that.